flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecTableSourceScan.java

```diff
@@ -131,15 +133,17 @@ public DynamicTableSourceSpec getTableSourceSpec() {
         } else if (provider instanceof DataStreamScanProvider) {
             Transformation<RowData> transformation =
                     ((DataStreamScanProvider) provider)
-                            .produceDataStream(createProviderContext(), env)
+                            .produceDataStream(
+                                    createProviderContext(planner.getConfiguration()), env)
```

Comment:
Use the new `ExecNodeConfig` here. `planner.getConfiguration()` should not be accessed directly anymore.